tfhe/src/core_crypto/commons/numeric/unsigned.rs

```diff
@@ -336,4 +368,37 @@ mod test {
             assert_eq!(expected_res, res, "a: {a}, b: {b}");
         }
     }
+
+    #[test]
+    fn test_wrapping_mul_custom_mod() {
+        let custom_modulus_u128 = (1u128 << 64) - (1 << 32) + 1;
+        let custom_modulus = custom_modulus_u128 as u64;
+        let a = u64::MAX % custom_modulus;
+        let b = u64::MAX % custom_modulus;
+
+        let a_u128: u128 = a.into();
+        let b_u128: u128 = b.into();
+
+        let expected_res = ((a_u128 * b_u128) % custom_modulus_u128) as u64;
+
+        let res = a.wrapping_mul_custom_mod(b, custom_modulus);
+        assert_eq!(expected_res, res);
+
+        const NB_REPS: usize = 100_000_000;
+
+        use rand::Rng;
+        let mut thread_rng = rand::thread_rng();
+        for _ in 0..NB_REPS {
+            let a = thread_rng.gen::<u64>() % custom_modulus;
+            let b = thread_rng.gen::<u64>() % custom_modulus;
```

Comment:
`custom_modulus` could also be taken randomly

Reply:
agreed, though we are interested in the Solinas prime at least as it's one that has been discussed at length, I'll add some other moduli and have a random one in there

Reply:
done